pkg/cloud/services/elb/loadbalancer.go

```diff
@@ -1567,3 +1567,15 @@ func chunkELBs(names []string) [][]string {
 	}
 	return chunked
 }
+
+func shouldReconcileSGs(lbType infrav1.LoadBalancerType, existingSGs []string, specSGs []string) bool {
+	// Backwards compat: NetworkLoadBalancers were not always capable of having security groups attached.
+	// Once created without a security group, the NLB can never have any added.
+	if lbType == infrav1.LoadBalancerTypeNLB && len(existingSGs) == 0 {
+		return false
```

Comment:
Users won't get a warning in logs if their desired manifest specifies SGs but their existing AWS NLB has none. Can we maybe add a log line for this?

Reply:
Done!